cmd/ipfs/daemon.go

```diff
@@ -301,15 +303,15 @@ func daemonFunc(req cmds.Request, res cmds.Response) {
 
 	routingOption, _, err := req.Option(routingOptionKwd).String()
 	if err != nil {
-		res.SetError(err, cmds.ErrNormal)
+		re.SetError(err, cmdsutil.ErrNormal)
 		return
 	}
 	switch routingOption {
 	case routingOptionSupernodeKwd:
 		servers, err := cfg.SupernodeRouting.ServerIPFSAddrs()
 		if err != nil {
-			res.SetError(err, cmds.ErrNormal)
-			repo.Close() // because ownership hasn't been transferred to the node
+			defer repo.Close() // because ownership hasn't been transferred to the node
```

Comment:
argh sorry didn't mean to delety your comment.

I defer because now SetError may panic

Reply:
When would SetError panic?